libnetwork/sandbox_linux.go

```
@@ -319,6 +319,12 @@ func (sb *Sandbox) populateNetworkResources(ctx context.Context, ep *Endpoint) e
 		if sysctls := ep.getSysctls(); len(sysctls) > 0 {
 			ifaceOptions = append(ifaceOptions, osl.WithSysctls(sysctls))
 		}
+		if nMsgs, ok := ep.advertiseAddrNMsgs(); ok {
+			ifaceOptions = append(ifaceOptions, osl.WithAdvertiseAddrNMsgs(nMsgs))
+		}
+		if interval, ok := ep.advertiseAddrInterval(); ok {
+			ifaceOptions = append(ifaceOptions, osl.WithAdvertiseAddrInterval(interval))
+		}
```

Both `advertiseAddrNMsgs` and `advertiseAddrInterval` appear to be wrappers for `network.XXX` (both do `ep.getNetwork()` as part of that).

Mostly wondering here; should we remove the extra indirection (fetch `ep.getNetwork()` once, then call these methods), or would that be too confusing?

I also noticed that both  `advertiseAddrNMsgs` and `advertiseAddrInterval`  perform validation of the values (but discarding the actual errors if any would be there); the same validation is already done on `Controller.NewNetwork()`, so curious if that state can be modified in between that requires these to be validated again. (and _if_ that's the case if validation should happen before we do all this, instead of those validation errors to be silently discarded).

---

> Both `advertiseAddrNMsgs` and `advertiseAddrInterval` appear to be wrappers for `network.XXX` (both do `ep.getNetwork()` as part of that).
> 
> Mostly wondering here; should we remove the extra indirection (fetch `ep.getNetwork()` once, then call these methods), or would that be too confusing?

Ok, I've removed the indirection.

> I also noticed that both `advertiseAddrNMsgs` and `advertiseAddrInterval` perform validation of the values (but discarding the actual errors if any would be there); the same validation is already done on `Controller.NewNetwork()`, so curious if that state can be modified in between that requires these to be validated again. (and _if_ that's the case if validation should happen before we do all this, instead of those validation errors to be silently discarded).

It deals with Linux vs. Windows ... the config has already been validated for Linux, but the Sandbox will call this for Windows too. I've changed the names `validateAdvertiseAddrXXX` to `validatedAdvertiseAddrXXX`, and added comments to try to make that clear.